File: api/Models/Enrollment.cs
namespace SkillMetrix_LMS.API.Models;

public class Enrollment
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public Guid CourseId { get; set; }
    public decimal PricePaid { get; set; }
    public DateTime EnrolledAt { get; set; } = DateTime.UtcNow;
    public User User { get; set; } = null!;
    public Course Course { get; set; } = null!;
}

File: client/src/types/enrollment.ts
// ─── Enrollment Types ─────────────────────────────────────────────────────────

export interface EnrollmentCourseInfo {
    id: string;
    title: string;
    thumbnail?: string;
    instructorName: string;
    chapterCount?: number;
    lessonCount?: number;
}

export interface EnrollmentDto {
    id: string;
    userId: string;
    courseId: string;
    pricePaid: number;
    enrolledAt: string;
    course?: EnrollmentCourseInfo;
}

export interface EnrollmentCourseProgress {
    enrollmentId: string;
    courseId: string;
    courseTitle: string;
    thumbnail?: string;
    instructorName: string;
    totalLessons: number;
    completedLessons: number;
    percentComplete: number;
    lastActivityAt?: string;
    enrolledAt: string;
}
